### PR TITLE
feat: adopt createMessage upload api

### PR DESCRIPTION
**Pull request by pengx17**

### CODE REVIEW

**Comment by pengx17** [comment]
* **#6596** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6596?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#6584** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/6584?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @pengx17 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/request.ts [hunk 1]
```diff
@@ -70,28 +65,32 @@ async function createSessionMessage({
...
+              return new File([blob], await calculateBlobHash(blob));
+            }
+          })
+      );
+      const stringAttachments = attachments.filter(
+        (attachment): attachment is string => typeof attachment === 'string'
       );
-      options.attachments = normalizedAttachments;
+      options.attachments = stringAttachments;
+      options.blobs = blobs;
```
Comment (darkskygit) [inline packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/request.ts]: consider https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/groupBy ?

Comment (pengx17) [reply packages/frontend/core/src/components/blocksuite/block-suite-editor/ai/request.ts]: It seems this api is too new. I will use groupBy from lodash instead

File: packages/backend/server/src/plugins/copilot/resolver.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ class CreateChatMessageInput implements Omit<SubmittedMessage, 'content'> {
   attachments!: string[] | undefined;
 
   @Field(() => [GraphQLUpload], { nullable: true })
-  blobs!: FileUpload[] | undefined;
+  blobs!: Promise<FileUpload>[] | undefined;
```
Comment (Brooooooklyn) [inline packages/backend/server/src/plugins/copilot/resolver.ts:67]: looks so weird

Comment (pengx17) [reply packages/backend/server/src/plugins/copilot/resolver.ts:67]: but it works lol

Comment (pengx17) [reply packages/backend/server/src/plugins/copilot/resolver.ts:67]: Seems relavent

**Review by forehalo — APPROVED** [review]

**Review by Brooooooklyn — APPROVED** [review]

**Review by darkskygit — APPROVED** [review]

**Pull request merged**